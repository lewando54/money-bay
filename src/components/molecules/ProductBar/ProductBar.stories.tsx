/* eslint-disable @typescript-eslint/consistent-type-assertions */
import { StoryObj, Meta } from '@storybook/react-native'

import ProductBar from './ProductBar'

const meta = {
    component: ProductBar,
    argTypes: {
        product: { control: 'text' },
        name: { control: 'text' },
        balance: { control: 'number' },
        balanceTextColor: { control: 'text' },
        cardIssuer: { control: 'select', options: ['visa', 'mastercard'] },
        currency: { control: 'object' },
        hasMenuButton: { control: 'boolean' }, 
        onMenuClick: { action: 'clickedOnMenu' }
    }
} satisfies Meta<typeof ProductBar>

export default meta
type Story = StoryObj<typeof ProductBar>

export const DefaultCard: Story = {
    args: {
        product: 'card',
        name: '**** **** **** 7895',
        balance: 2500,
        cardIssuer: 'visa',
        currency: {iso: 'usd', symbol: '$', left: true},
        hasMenuButton: false
    }
}

export const DefaultAccount: Story = {
    args: {
        product: 'account',
        name: 'US**********************4571',
        balance: 39863.62,
        cardIssuer: 'visa',
        currency: {iso: 'usd', symbol: '$', left: true},
        hasMenuButton: false
    }
}

export const CustomBalanceColor: Story = {
    args: {
        product: 'card',
        name: '**** **** **** 7895',
        balance: 2500,
        balanceTextColor: '#FF5887',
        cardIssuer: 'visa',
        currency: {iso: 'usd', symbol: '$', left: true},
        hasMenuButton: false
    }
}

export const HasMenuButton: Story = {
    args: {
        product: 'card',
        name: '**** **** **** 7895',
        balance: 2500,
        cardIssuer: 'visa',
        currency: {iso: 'usd', symbol: '$', left: true},
        hasMenuButton: true
    }
}
