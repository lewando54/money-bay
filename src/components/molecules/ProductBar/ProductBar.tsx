import React from 'react'
import { Image, Text, View } from 'react-native'
import { Entypo } from '@expo/vector-icons' 

import { MiniCreditCard, UniversalContainer } from '@atoms'
import { TCardIssuer } from '@utils/cardIssuers'
import { TCurrency } from '@utils/currency'

import productBarStyle from './ProductBar.style'

import globalThemeStyle, { globalColors } from '@styling/GlobalTheme.style'

type TProduct = 'card' | 'account'

interface IProductBarProps {
    product?: TProduct
    name?: string
    balance?: number
    balanceTextColor?: string
    cardIssuer?: TCardIssuer
    currency?: TCurrency
    hasMenuButton?: boolean
    onMenuClick?: () => void
}

export default function ProductBar({
    product='card',
    name='**** **** **** 7895',
    balance=2500,
    balanceTextColor,
    cardIssuer='visa',
    currency={iso: 'usd', symbol: '$', left: true},
    hasMenuButton=false, 
    onMenuClick,
}: IProductBarProps){
    const ProductIcon = 
        product == 'card' ? 
            <MiniCreditCard cardIssuer={cardIssuer}/> :
            product == 'account' &&
            <Image source={require('./assets/check.png')} />

    const ThreeDotsMenu = hasMenuButton ? <Entypo name="dots-three-vertical" size={16} color={globalColors.BODY_TEXT_COLOR} testID='three-dots-button' style={{marginLeft: 'auto'}} onPress={onMenuClick}/> : undefined

    return (
        <UniversalContainer style={productBarStyle.container}>
            {ProductIcon}
            <View>
                <Text style={[globalThemeStyle.text_Regular, {color: globalColors.BODY_TEXT_COLOR}]}>
                    {name}
                </Text>
                <Text style={[globalThemeStyle.text_SemiBold, {color: balanceTextColor ? balanceTextColor : globalColors.MAIN_DARK}]}>
                    {balance.toLocaleString('en-US', {minimumIntegerDigits: 2, minimumFractionDigits: 2}).replace(',', ' ')} {currency.iso.toLocaleUpperCase()}
                </Text>
            </View>
            {ThreeDotsMenu}
        </UniversalContainer>
    )
}
