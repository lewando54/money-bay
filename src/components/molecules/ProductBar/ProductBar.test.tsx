import React from 'react'
import renderer from 'react-test-renderer'
import { fireEvent, render, screen } from '@testing-library/react-native'
import ProductBar from './ProductBar'
describe('ProductBar', () => {
  it('should render correctly', () => {
    const tree = renderer.create(<ProductBar />)
    expect(tree).toMatchSnapshot()
  })

  it('should render minicreditcard when passed card product', () => {
    const tree = renderer.create(<ProductBar product='card'/>)
    expect(tree).toMatchSnapshot()
  })

  it('should render account icon when passed account product', () => {
    const tree = renderer.create(<ProductBar product='account'/>)
    expect(tree).toMatchSnapshot()
  })

  it('should render without menu button', () => {
    const tree = renderer.create(<ProductBar hasMenuButton={false}/>)
    expect(tree).toMatchSnapshot()
  })

  it('should render with menu button', () => {
    const tree = renderer.create(<ProductBar hasMenuButton={true}/>)
    expect(tree).toMatchSnapshot()
  })

  it('should render custom balanceTextColor', () => {
    const tree = renderer.create(<ProductBar balanceTextColor='#fff'/>)
    expect(tree).toMatchSnapshot()
  })

  it('should call onMenuClick when menu button clicked', () => {
    const onClick = jest.fn()
    const { getByTestId } = render(<ProductBar onMenuClick={onClick} hasMenuButton={true}/>)
    fireEvent.press(getByTestId('three-dots-button'))
    expect(onClick).toBeCalled()
  })
})
