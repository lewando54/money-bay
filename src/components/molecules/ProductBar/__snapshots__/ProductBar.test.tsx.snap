// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductBar should render account icon when passed account product 1`] = `
<View
  style={
    [
      {
        "backgroundColor": "white",
        "borderRadius": 10,
        "height": "auto",
        "padding": 10,
        "width": "100%",
      },
      false,
      {
        "alignItems": "center",
        "flexDirection": "row",
        "gap": 10,
      },
    ]
  }
  testID="universal-container"
>
  <Image
    source={1}
  />
  <View>
    <Text
      style={
        [
          {
            "color": "#1B1D4D",
            "fontFamily": "MulishRegular",
            "fontSize": 14,
          },
          [
            {
              "color": "white",
              "fontFamily": "MulishRegular",
            },
            {
              "color": "#4C4C60",
            },
          ],
        ]
      }
    >
      **** **** **** 7895
    </Text>
    <Text
      style={
        [
          {
            "color": "#1B1D4D",
            "fontFamily": "MulishRegular",
            "fontSize": 14,
          },
          [
            {
              "color": "white",
              "fontFamily": "MulishSemiBold",
            },
            {
              "color": "#1B1D4D",
            },
          ],
        ]
      }
    >
      2 500.00
       
      USD
    </Text>
  </View>
</View>
`;

exports[`ProductBar should render correctly 1`] = `
<View
  style={
    [
      {
        "backgroundColor": "white",
        "borderRadius": 10,
        "height": "auto",
        "padding": 10,
        "width": "100%",
      },
      false,
      {
        "alignItems": "center",
        "flexDirection": "row",
        "gap": 10,
      },
    ]
  }
  testID="universal-container"
>
  <ViewManagerAdapter_ExpoLinearGradient
    proxiedProperties={
      {
        "colors": [
          4290938283,
          4286226894,
          4279967594,
        ],
        "endPoint": [
          1,
          1,
        ],
        "locations": undefined,
        "startPoint": [
          0,
          0,
        ],
      }
    }
    style={
      {
        "borderRadius": 5,
        "height": 46,
        "padding": 10,
        "width": 72,
      }
    }
  >
    <Image
      source={1}
      style={
        {
          "height": 15,
          "marginLeft": "auto",
          "marginTop": "auto",
          "resizeMode": "contain",
          "width": 25,
        }
      }
    />
  </ViewManagerAdapter_ExpoLinearGradient>
  <View>
    <Text
      style={
        [
          {
            "color": "#1B1D4D",
            "fontFamily": "MulishRegular",
            "fontSize": 14,
          },
          [
            {
              "color": "white",
              "fontFamily": "MulishRegular",
            },
            {
              "color": "#4C4C60",
            },
          ],
        ]
      }
    >
      **** **** **** 7895
    </Text>
    <Text
      style={
        [
          {
            "color": "#1B1D4D",
            "fontFamily": "MulishRegular",
            "fontSize": 14,
          },
          [
            {
              "color": "white",
              "fontFamily": "MulishSemiBold",
            },
            {
              "color": "#1B1D4D",
            },
          ],
        ]
      }
    >
      2 500.00
       
      USD
    </Text>
  </View>
</View>
`;

exports[`ProductBar should render custom balanceTextColor 1`] = `
<View
  style={
    [
      {
        "backgroundColor": "white",
        "borderRadius": 10,
        "height": "auto",
        "padding": 10,
        "width": "100%",
      },
      false,
      {
        "alignItems": "center",
        "flexDirection": "row",
        "gap": 10,
      },
    ]
  }
  testID="universal-container"
>
  <ViewManagerAdapter_ExpoLinearGradient
    proxiedProperties={
      {
        "colors": [
          4290938283,
          4286226894,
          4279967594,
        ],
        "endPoint": [
          1,
          1,
        ],
        "locations": undefined,
        "startPoint": [
          0,
          0,
        ],
      }
    }
    style={
      {
        "borderRadius": 5,
        "height": 46,
        "padding": 10,
        "width": 72,
      }
    }
  >
    <Image
      source={1}
      style={
        {
          "height": 15,
          "marginLeft": "auto",
          "marginTop": "auto",
          "resizeMode": "contain",
          "width": 25,
        }
      }
    />
  </ViewManagerAdapter_ExpoLinearGradient>
  <View>
    <Text
      style={
        [
          {
            "color": "#1B1D4D",
            "fontFamily": "MulishRegular",
            "fontSize": 14,
          },
          [
            {
              "color": "white",
              "fontFamily": "MulishRegular",
            },
            {
              "color": "#4C4C60",
            },
          ],
        ]
      }
    >
      **** **** **** 7895
    </Text>
    <Text
      style={
        [
          {
            "color": "#1B1D4D",
            "fontFamily": "MulishRegular",
            "fontSize": 14,
          },
          [
            {
              "color": "white",
              "fontFamily": "MulishSemiBold",
            },
            {
              "color": "#fff",
            },
          ],
        ]
      }
    >
      2 500.00
       
      USD
    </Text>
  </View>
</View>
`;

exports[`ProductBar should render minicreditcard when passed card product 1`] = `
<View
  style={
    [
      {
        "backgroundColor": "white",
        "borderRadius": 10,
        "height": "auto",
        "padding": 10,
        "width": "100%",
      },
      false,
      {
        "alignItems": "center",
        "flexDirection": "row",
        "gap": 10,
      },
    ]
  }
  testID="universal-container"
>
  <ViewManagerAdapter_ExpoLinearGradient
    proxiedProperties={
      {
        "colors": [
          4290938283,
          4286226894,
          4279967594,
        ],
        "endPoint": [
          1,
          1,
        ],
        "locations": undefined,
        "startPoint": [
          0,
          0,
        ],
      }
    }
    style={
      {
        "borderRadius": 5,
        "height": 46,
        "padding": 10,
        "width": 72,
      }
    }
  >
    <Image
      source={1}
      style={
        {
          "height": 15,
          "marginLeft": "auto",
          "marginTop": "auto",
          "resizeMode": "contain",
          "width": 25,
        }
      }
    />
  </ViewManagerAdapter_ExpoLinearGradient>
  <View>
    <Text
      style={
        [
          {
            "color": "#1B1D4D",
            "fontFamily": "MulishRegular",
            "fontSize": 14,
          },
          [
            {
              "color": "white",
              "fontFamily": "MulishRegular",
            },
            {
              "color": "#4C4C60",
            },
          ],
        ]
      }
    >
      **** **** **** 7895
    </Text>
    <Text
      style={
        [
          {
            "color": "#1B1D4D",
            "fontFamily": "MulishRegular",
            "fontSize": 14,
          },
          [
            {
              "color": "white",
              "fontFamily": "MulishSemiBold",
            },
            {
              "color": "#1B1D4D",
            },
          ],
        ]
      }
    >
      2 500.00
       
      USD
    </Text>
  </View>
</View>
`;

exports[`ProductBar should render with menu button 1`] = `
<View
  style={
    [
      {
        "backgroundColor": "white",
        "borderRadius": 10,
        "height": "auto",
        "padding": 10,
        "width": "100%",
      },
      false,
      {
        "alignItems": "center",
        "flexDirection": "row",
        "gap": 10,
      },
    ]
  }
  testID="universal-container"
>
  <ViewManagerAdapter_ExpoLinearGradient
    proxiedProperties={
      {
        "colors": [
          4290938283,
          4286226894,
          4279967594,
        ],
        "endPoint": [
          1,
          1,
        ],
        "locations": undefined,
        "startPoint": [
          0,
          0,
        ],
      }
    }
    style={
      {
        "borderRadius": 5,
        "height": 46,
        "padding": 10,
        "width": 72,
      }
    }
  >
    <Image
      source={1}
      style={
        {
          "height": 15,
          "marginLeft": "auto",
          "marginTop": "auto",
          "resizeMode": "contain",
          "width": 25,
        }
      }
    />
  </ViewManagerAdapter_ExpoLinearGradient>
  <View>
    <Text
      style={
        [
          {
            "color": "#1B1D4D",
            "fontFamily": "MulishRegular",
            "fontSize": 14,
          },
          [
            {
              "color": "white",
              "fontFamily": "MulishRegular",
            },
            {
              "color": "#4C4C60",
            },
          ],
        ]
      }
    >
      **** **** **** 7895
    </Text>
    <Text
      style={
        [
          {
            "color": "#1B1D4D",
            "fontFamily": "MulishRegular",
            "fontSize": 14,
          },
          [
            {
              "color": "white",
              "fontFamily": "MulishSemiBold",
            },
            {
              "color": "#1B1D4D",
            },
          ],
        ]
      }
    >
      2 500.00
       
      USD
    </Text>
  </View>
  <Text />
</View>
`;

exports[`ProductBar should render without menu button 1`] = `
<View
  style={
    [
      {
        "backgroundColor": "white",
        "borderRadius": 10,
        "height": "auto",
        "padding": 10,
        "width": "100%",
      },
      false,
      {
        "alignItems": "center",
        "flexDirection": "row",
        "gap": 10,
      },
    ]
  }
  testID="universal-container"
>
  <ViewManagerAdapter_ExpoLinearGradient
    proxiedProperties={
      {
        "colors": [
          4290938283,
          4286226894,
          4279967594,
        ],
        "endPoint": [
          1,
          1,
        ],
        "locations": undefined,
        "startPoint": [
          0,
          0,
        ],
      }
    }
    style={
      {
        "borderRadius": 5,
        "height": 46,
        "padding": 10,
        "width": 72,
      }
    }
  >
    <Image
      source={1}
      style={
        {
          "height": 15,
          "marginLeft": "auto",
          "marginTop": "auto",
          "resizeMode": "contain",
          "width": 25,
        }
      }
    />
  </ViewManagerAdapter_ExpoLinearGradient>
  <View>
    <Text
      style={
        [
          {
            "color": "#1B1D4D",
            "fontFamily": "MulishRegular",
            "fontSize": 14,
          },
          [
            {
              "color": "white",
              "fontFamily": "MulishRegular",
            },
            {
              "color": "#4C4C60",
            },
          ],
        ]
      }
    >
      **** **** **** 7895
    </Text>
    <Text
      style={
        [
          {
            "color": "#1B1D4D",
            "fontFamily": "MulishRegular",
            "fontSize": 14,
          },
          [
            {
              "color": "white",
              "fontFamily": "MulishSemiBold",
            },
            {
              "color": "#1B1D4D",
            },
          ],
        ]
      }
    >
      2 500.00
       
      USD
    </Text>
  </View>
</View>
`;
